### PR TITLE
arc: Add version 3.5.0

### PR DESCRIPTION
**Pull request by KaranKad**
Added arc 
Archiver and extractor cli software

Closes https://github.com/ScoopInstaller/Main/issues/3889

Relates to #XXXX

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rashil2000** [comment]
For `checkver`, use the Releases page as `url` and downloaded filename as `regex`.

**Comment by KaranKad** [comment]
Need some help with regex

File: bucket/arc.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        },
+        "32bit": {
+            "url": "https://github.com/mholt/archiver/releases/download/v3.5.0/arc_3.5.0_windows_386.exe#/arc.exe",
+            "hash": "sha256:ac5908fe97f62d042f82196661d6f9a9c3cf85eeb30eb4cea877a2fefde599b3"
+        }
+    },
+    "bin": "arc.exe",
+    "checkver": {
+        "url": "https://github.com/mholt/archiver/releases",
+        "regex": "arc_([\\d]+)_windows_amd64\\.exe"
```
Comment (rashil2000) [inline bucket/arc.json:19]:
```suggestion
        "regex": "arc_([\\d.]+)_windows_amd64\\.exe"
```

**Pull request merged**